guides/micronaut-kotlin-extension-fns/kotlin/src/test/kotlin/example/micronaut/DadJokeTest.kt

```diff
@@ -0,0 +1,34 @@
+package example.micronaut
+import io.micronaut.context.createBean
+import io.micronaut.context.run
+import io.micronaut.http.HttpRequest
+import io.micronaut.http.client.HttpClient
+import io.micronaut.http.retrieveList
+import io.micronaut.http.retrieveObject
+import io.micronaut.runtime.server.EmbeddedServer
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Test
+
+@MicronautTest
+class DadJokeTest {
+
+    @Test
+    fun testDadJokeController() {
+        val embeddedServer = run<EmbeddedServer>() // <1>
+        val client = embeddedServer.applicationContext.createBean<HttpClient>(embeddedServer.url).toBlocking() // <2>
```

Comment:
As you are using `@MicronautTest` you don't need to start the embedded server or create the httpclient bean. Those are available in the test and you can just `@Inject` them.
As you want to show the usage of extension functions you should remove `@MicronautTest`